extractor/extractor/filing.py

```diff
@@ -0,0 +1,135 @@
+from __future__ import annotations
+import logging
+from lxml import etree
+import sys
+from typing import Dict, Optional, TextIO
+
+
+_logger = logging.getLogger(__name__)
+
+
+class Filing:
+    """
+    Encapsulate a Filing as a semantic object with appropriately named
+    fields.
+
+    The implementation actually returned from Result objects will have its
+    fields generated at build time to avoid the need to update the code every
+    time someone wants to access a new field.
+
+    Fields that aren't included can still be accessed using the XPath strings
+    listed on http://www.irsx.info/ and the `get_*` methods. The paths should
+    be used verbatim, in their absolute form, without namespaces.
+
+    >>> with open('fixtures/filing.xml') as xml:
+    ...    filing = Filing(xml)
+    >>> filing.formation_year
+    '2004'
+    >>> filing.principal_officer_name
+    'SCOTT LEWIS'
+    """
+
+    _namespaces: Dict[str, str]
+
+    _tree: etree.ElementTree
+
+    def __init__(
+        self, xml_file: TextIO,
+    ):
+        self._namespaces = {
+            "efile": "http://www.irs.gov/efile",
+        }
+
+        content = xml_file.read()
+        while not content.startswith("<"):
+            content = content[1:]
+        content = content.encode("utf-8")
+        _logger.info(f"XML: '{content[:40]}'")
```

Comment:
what's at the 40th index?

Reply:
Nothing in particular, just trying to get a little bit of the text but without barfing too much into the terminal.